packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -784,6 +846,7 @@ class TextStyle extends Diagnosticable {
     styles.add(new EnumProperty<TextBaseline>('${prefix}baseline', textBaseline, defaultValue: null));
     styles.add(new DoubleProperty('${prefix}height', height, unit: 'x', defaultValue: null));
     styles.add(new StringProperty('${prefix}locale', locale?.toString(), defaultValue: null, quoted: false));
+    styles.add(new StringProperty('${prefix}foreground', foreground?.toString(), defaultValue: null, quoted: false));
     styles.add(new StringProperty('${prefix}background', background?.toString(), defaultValue: null, quoted: false));
```

Comment:
for locale, foreground, and background, we should use `DiagnosticsProperty` with an appropriate type, e.g. `DiagnosticsProperty<Paint>`.